### PR TITLE
Feature/GitHub actions

### PR DESCRIPTION
**Pull request by ikrivosheev**
@mdsteele, hello! Thank you for the library. I add simple pipeline for linters and tests for the library and fix clippy warnings.

You can see pipeline: https://github.com/ikrivosheev/rust-msi/actions/runs/1986369276

### CODE REVIEW

**Comment by ikrivosheev** [comment]
@mdsteele I fix work with time, pipeline: https://github.com/ikrivosheev/rust-msi/actions/runs/2001281343

File: examples/msiquery.rs [hunk 1]
```diff
@@ -7,6 +7,7 @@ extern crate pest_derive;
 use clap::{App, Arg};
 use msi::{Delete, Expr, Insert, Select, Update, Value};
 use pest::Parser;
+use std::ops::Not;
```
Comment (mdsteele) [inline examples/msiquery.rs:10]: Is this line needed?

File: src/internal/expr.rs [hunk 2]
```diff
@@ -144,13 +135,26 @@ impl Expr {
     }
 }
 
+/// Returns an expression that evaluates to true if the subexpression
+/// evaluates to false.
+///
+/// This method exists instead of the `std::ops::Not` trait to distinguish
```
Comment (mdsteele) [inline src/internal/expr.rs:141]: `Expr` intentionally doesn't implement `std::ops::Not`, because Rust uses that for both logical and bitwise not, but the `Expr` type needs to distinguish those two operations.

File: src/internal/time.rs [hunk 3]
```diff
@@ -6,7 +6,7 @@ use std::time::{Duration, SystemTime, UNIX_EPOCH};
 pub fn filetime_from_system_time(timestamp: SystemTime) -> u64 {
     match timestamp.duration_since(filetime_epoch()) {
         Ok(delta) => {
-            delta.as_secs() * 10_000_000 + (delta.subsec_nanos() / 100) as u64
+            delta.as_secs() * 1_000_000_000 + delta.subsec_nanos() as u64
```
Comment (mdsteele) [inline src/internal/time.rs:9]: Why these changes?

**Comment by mdsteele** [comment]
Thanks for the PR!  Just a few questions above.

File: src/internal/time.rs (hunk 3 above)
Comment (ikrivosheev) [reply src/internal/time.rs:9]: Yes, because the tests are falling: https://github.com/ikrivosheev/rust-msi/runs/5552772525?check_suite_focus=true#step:4:207

`nanos` is 10^(-9)

Comment (ikrivosheev) [reply src/internal/time.rs:9]: I can rewrite tests and revert this changes?

File: src/internal/expr.rs (hunk 2 above)
Comment (ikrivosheev) [reply src/internal/expr.rs]: Thank you, I understand. Done

File: examples/msiquery.rs (hunk 1 above)
Comment (ikrivosheev) [reply examples/msiquery.rs]: Done

File: src/internal/time.rs (hunk 3 above)
Comment (mdsteele) [reply src/internal/time.rs:9]: Ah, I think I understand.

Windows [FILETIME](https://docs.microsoft.com/en-us/windows/win32/api/minwinbase/ns-minwinbase-filetime)s are measured in 10-million-ths of a second, that is, in 100-nanosecond intervals.  So calculating the filetime as `delta.as_secs() * 10_000_000 + (delta.subsec_nanos() / 100)` is intentional.

Looks like the failing test is checking that `PropertyValue::FileTime(SystemTime::now())` can round-trip, and that's failing on (ubuntu-20.04, stable) with:
```
---- internal::propset::tests::property_value_round_trip stdout ----
thread 'internal::propset::tests::property_value_round_trip' panicked at 'assertion failed: `(left == right)`
  left: `FileTime(SystemTime { tv_sec: 1647343851, tv_nsec: 381626500 })`,
 right: `FileTime(SystemTime { tv_sec: 1647343851, tv_nsec: 381626523 })`', src/internal/propset.rs:508:13
```

I think the problem is that a Windows `FILETIME` can only represent a 100-nanosecond resolution, but the `SystemTime` clock on that OS has a higher resolution than that, so of course it can't always round-trip perfectly.  So the test itself is flawed.

Instead of using `SystemTime::now()` and expecting it to round-trip perfectly, I think we should either be using some specific `SystemTime` that is a multiple of 100 nanoseconds since the epoch, or else writing a different test that makes sure that the `PropertyValue::FileTime` round-trips to within 100 nanoseconds.

Comment (ikrivosheev) [reply src/internal/time.rs]: Done

**Comment by ikrivosheev** [comment]
@mdsteele I have done, can you review, please.

**Comment by mdsteele** [comment]
Thanks very much!
